### PR TITLE
Gradle: allow GnuPG for signing

### PR DESCRIPTION
**Pull request by miurahr**
It is convenient for developer who use GnuPG and want to publish OmegaT to localMaven to develop plugins.
It is even safer, because Gradle ask password interactively when running publish, and no need to write key password on property file.

Signed-off-by: Hiroshi Miura <miurahr@linux.com>

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1074,8 +1074,12 @@ publishing {
 
 signing {
     sign publishing.publications.mavenJava
+    if (!findProperty("signing.keyId") || findProperty("signing.keyId").equals("")) {
```
Comment (amake) [inline build.gradle:1077]:
```suggestion
    if (!findProperty("signing.keyId")) {
```

In Groovy, both null and the empty string are falsy.

File: build.gradle [hunk 2]
```diff
@@ -1074,8 +1074,12 @@ publishing {
...
+        useGpgCmd()
+    }
 }
 
+
```
Comment (amake) [inline build.gradle:1082]: This extra line can be reverted.

File: local.properties.example [hunk 3]
```diff
@@ -41,6 +41,10 @@ signing.keyId=
 signing.password=
 # Exported as e.g. `gpg --export-secret-keys > secring.gpg`
 signing.secretKeyRingFile=
+## When you want to use GnuPG command instead of exported keyring
+## uncommented and add key ID
+#signing.gnupg.keyName=
+#signing.gnupg.executable=gpg
```
Comment (amake) [inline local.properties.example:47]: I think we should improve the comments to explain better what is required and used for what.

Are `signing.{keyId,password,secretKeyRingFile}` used at all when configured with `useGpgCmd()`?

Comment (miurahr) [reply local.properties.example:47]: > Are signing.{keyId,password,secretKeyRingFile} used at all when configured with useGpgCmd()?

Yes.

These properties are documented at [gradle manual](https://docs.gradle.org/current/userguide/signing_plugin.html#sec:sec:gnupg_signatory_configuration)

Comment (miurahr) [reply local.properties.example]: I've updated comments.

**Comment by amake** [comment]
As far as I can tell, it seems like `useGpgCmd()` doesn't use any of the regular `signing.{keyId,password,secretKeyRingFile}` properties; it only uses the `signing.gnupg.*` properties.

Since there are so many `signing.gnupg.*` properties and using gpg-agent seems like a bit of a power-user option, I simplified the comments a bit.

Let me know if this is OK for you.

**Comment by miurahr** [comment]
> using gpg-agent seems like a bit of a power-user option

It may be a true, I don't know.Using Java signatory is very weird  for me who has experience Using PGP and GnuPG in past 30 years. :-)
Yes, I'm power-user. It is enough comment.

**Comment by amake** [comment]
Thanks. I've squashed to `master`.

**Pull request closed without merging**